File: frontend/src/api/buscar.ts
import api from "./client";

export interface PaqueteItem {
  clave: string;
  tipo: "serial" | "orden";
  numero_orden: string | null;
  cliente: string | null;
  mensajero: string | null;
  ciudad: string | null;
  fecha: string | null;
  estado: string;
  planilla: string | null;
  tipo_gestion: string | null;
}

export interface BuscarResultado {
  total: number;
  seriales: number;
  ordenes: number;
  items: PaqueteItem[];
}

export type BuscarModo = "serial" | "orden" | "cliente";

export const buscarApi = {
  buscarPaquete: (q: string, modo: BuscarModo) =>
    api.get<BuscarResultado>("/buscar/paquete", { params: { q, modo } }),
};

File: backend/app/schemas/buscar.py
from pydantic import BaseModel


class PaqueteItem(BaseModel):
    clave: str
    tipo: str
    numero_orden: str | None = None
    cliente: str | None = None
    mensajero: str | None = None
    ciudad: str | None = None
    fecha: str | None = None
    estado: str
    planilla: str | None = None
    tipo_gestion: str | None = None


class BuscarResultado(BaseModel):
    total: int
    seriales: int
    ordenes: int
    items: list[PaqueteItem]
